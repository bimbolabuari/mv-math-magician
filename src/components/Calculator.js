import React from "react";
import calculate from "../logic/Calculate";
import DisplayResult from "./DisplayResult";
import CalculatorButton from "./CalculatorButton";
import generateDisplayValue from '../helper/generateDisplayValue';

class Calculator extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  clickEventListener = (event) => {
    const object = this.state;
    const value = event.target.value;
    const result = calculate(object, value);
    this.setState(result);
  }

  render() {
    const displayValue = generateDisplayValue(this.state);
    return (
      <div className="calculator">
      <DisplayResult displayvalue={displayValue}/>
      <div className="calculator-keys"> 
        <CalculatorButton  value="AC"  clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="+/-" clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="%"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="รท"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="7"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="8"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="9"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="x"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="4"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="5"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="6"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="-"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="1"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="2"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="3"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="+"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="0"   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="."   clickEventListener={this.clickEventListener}/>
        <CalculatorButton  value="="   clickEventListener={this.clickEventListener}/>
      </div>
    </div>
    );
  }
}

  
  export default Calculator;
  