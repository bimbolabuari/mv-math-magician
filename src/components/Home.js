import React from "react";

 const Home = () => {
    return (
        <div>
            <h2 className="homeTitle">Welcome to our page</h2>
            <p className="homeParagraph">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id nisi erat. Suspendisse convallis erat non orci auctor, ut venenatis elit tincidunt.
             <br/>Ut non augue non metus interdum ultricies vel quis felis. Vivamus vitae sapien ut elit blandit faucibus. Nulla a nunc eu lacus mollis congue. <br/>Nulla facilisi. Nam porttitor tellus vestibulum sem sodales hendrerit. Quisque pulvinar urna vitae nisi dignissim, in lacinia urna laoreet.</p>
        </div>
    )
}

export default Home;