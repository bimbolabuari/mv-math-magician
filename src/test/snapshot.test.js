import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Tests snapshots of components', () => {
  it('renders correctly', () => {
    const UICalculator = renderer
      .create(<Calculator />)
      .toJSON();
    expect(UICalculator).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const UIHome = renderer
      .create(<Home />)
      .toJSON();
    expect(UIHome).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const UIQuote = renderer
      .create(<Quote />)
      .toJSON();
    expect(UIQuote).toMatchSnapshot();
  });
});