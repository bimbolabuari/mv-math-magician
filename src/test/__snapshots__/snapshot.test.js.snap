// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests snapshots of components renders correctly 1`] = `
<div>
  <h2
    className="calculatorTitle"
  >
    Let\`s do some Maths!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="calculator-screen"
    >
      0
    </div>
    <div
      className="calculator-keys"
    >
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className=" operator"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className=" operator"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className=" operator"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className=" operator"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="zero "
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className=" "
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className=" operator"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Tests snapshots of components renders correctly 2`] = `
<div>
  <h2
    className="homeTitle"
  >
    Welcome to our page
  </h2>
  <p
    className="homeParagraph"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id nisi erat. Suspendisse convallis erat non orci auctor, ut venenatis elit tincidunt.
    <br />
    Ut non augue non metus interdum ultricies vel quis felis. Vivamus vitae sapien ut elit blandit faucibus. Nulla a nunc eu lacus mollis congue. 
    <br />
    Nulla facilisi. Nam porttitor tellus vestibulum sem sodales hendrerit. Quisque pulvinar urna vitae nisi dignissim, in lacinia urna laoreet.
  </p>
</div>
`;

exports[`Tests snapshots of components renders correctly 3`] = `
<p
  className="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: It\`s about understanding. - Williams Paul
</p>
`;
